components/CreateCycle/CreateCycle.jsx

```diff
@@ -50,20 +53,27 @@ export default ({ navigation }) => {
 
   const createNewCycle = async (newCycle) => {
     const currentUser = firebase.auth().currentUser.uid;
-
     const cycleRef = firebase.firestore()
       .collection('users')
       .doc(currentUser)
       .collection('cycles');
-
-    const cycleDoc = await cycleRef.add(newCycle);
-
-    dispatch(actions.cycles.addCycle({
-      workouts: newCycle.workoutIds,
-      name: newCycle.name,
-      id: cycleDoc.id,
-    }));
-    navigation.goBack();
+    if (isNewCycle) {
+      const cycleDoc = await cycleRef.add(newCycle);
+      dispatch(actions.cycles.addCycle({
+        workouts: newCycle.workoutIds,
+        name: newCycle.name,
+        id: cycleDoc.id,
+      }));
+    } else {
+      await cycleRef.doc(cycleID).update(newCycle);
+      dispatch(actions.cycles.updateCycle(cycleID,
+        {
+          id: cycleID,
+          workouts: newCycle.workoutIds,
+          name: newCycle.name,
+        }));
+    }
+    navigation.navigate('Cycles');
```

Comment:
does navigaion.goBack() not work here? I think we should use that instead if it works here

Reply:
Fixed